### PR TITLE
[hail] Lower ArrayAgg/ArrayAggScan in Compile

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by akotlar** [comment]
15 lines * 11 :)

File: hail/src/main/scala/is/hail/expr/ir/lowering/LoweringPass.scala [hunk 1]
```diff
@@ -54,4 +56,40 @@ case object LowerTableToDistributedArrayPass extends LoweringPass {
...
+case object LowerArrayAggsToRunAggs extends LoweringPass {
+  val before: IRState = CompilableIR
+  val after: IRState = EmittableIR
+  val context: String = "LowerArrayAggsToRunAggs"
+
+  def transform(ctx: ExecuteContext, ir: BaseIR): BaseIR = RewriteBottomUp(ir, {
+    case ArrayAgg(a, name, query) =>
+      val res = genUID()
+      val aggs = Extract(query, res)
+      Some(Let(
```
Comment (akotlar) [inline hail/src/main/scala/is/hail/expr/ir/lowering/LoweringPass.scala]: Neat that we can rewrite the previous code (found in Emit) this tersely by using our existing primitives

**Comment by tpoterba** [comment]
The only code that should overlap with yours is in Compile, which has a 12-line diff! That's mostly what I was referring to.

**Comment by akotlar** [comment]
I can't easily tell whether the change to emitted code does not change behavior, looked a bit, so will wait for tests, but besides that, looks good

**Comment by tpoterba** [comment]
yeah, we have loads of tests go through this path. If Python tests pass, I'm quite satisfied this is correct!

**Comment by akotlar** [comment]
Same **** I'm dealing with:

2020-02-07 19:08:43 Hail: INFO: Exception:
is.hail.utils.HailException: not a streamable IR: (ArrayAggScan foo

**Comment by tpoterba** [comment]
for that one, we need to change the suite to call Compile or something. It's directly calling emit.

**Comment by akotlar** [comment]
Nice job!

**Pull request merged**